schemafy_lib/src/lib.rs

```diff
@@ -571,7 +689,23 @@ impl<'r> Expander<'r> {
             None => self.expand_definitions(schema),
         }
 
-        let types = self.types.iter().map(|t| &t.1);
+        // Dedupe because we may have visited the same file multiple times when de-referencing
+        // external files
```

Comment:
Shouldn't this be avoided altogether? If we have already visited a file it should be cached and not visited again

Reply:
Good eye. I was going to go back and revisit this but forgot.

Had to do some refactoring/bug-fixing to get this fixed, but the short version is that while we don't visit the same file twice, when we kick off a child Expander for a _new_ file, we shove all the types and Schemas from the parent Expander to it (to prevent unnecessary resolving.

when we merge the resolved types from the child Expander back into the parent Expander, we'd also get the same types that came from the Parent originally. I've updated `types` to be a HashMap to get around this and added a `sort` by type name before generating the AST to keep things deterministic.